File: 11 - Movie Review System/movie-review-frontend/src/api/reviewService.js
import api from "./api";

export const addReview = (movieId, review) =>
    api.post(`/reviews/${movieId}`, review);

export const getReviews = (movieId) =>
    api.get(`/reviews/${movieId}`);

export const getAverageRating = (movieId) =>
    api.get(`/reviews/${movieId}/average`);

export const deleteReview = (reviewId) =>
    api.delete(`/reviews/${reviewId}`);

File: 11 - Movie Review System/moviereview/src/main/java/com/dhanush/moviereview/service/ReviewService.java
package com.dhanush.moviereview.service;

import com.dhanush.moviereview.dto.ReviewRequest;
import com.dhanush.moviereview.entity.Review;

import java.util.List;

public interface ReviewService {

    Review addReview(Long movieId, ReviewRequest request);

    List<Review> getReviewsByMovie(Long movieId);

    void deleteReview(Long reviewId);

    Double getAverageRating(Long movieId);

}